packages/asset-swapper/src/utils/market_operation_utils/path_optimizer.ts

```diff
@@ -340,6 +342,12 @@ function findRoutesAndCreateOptimalPath(
     const vipSourcesRustRoute = new Float64Array(rustArgs.pathsIn.length);
     const vipSourcesOutputAmounts = new Float64Array(rustArgs.pathsIn.length);
 
+    // Get rustArgs samples for test purpose
+    if (SHOULD_ENABLE_RUST_DATA_SAMPLER) {
```

Comment:
I know it was like this in the initial spec, but I wonder if it's better to do the `Math.random() < SAMPLE_THRESHOLD` check here instead, saves us some execution for the vast majority of the cases where we do nothing
```suggestion
    if (SHOULD_ENABLE_RUST_DATA_SAMPLER && Math.random() < SAMPLE_THRESHOLD) {
```